src/test/java/com/znsio/sample/e2e/businessLayer/indigo/GiftVoucherBL.java

```diff
@@ -0,0 +1,107 @@
+public class GiftVoucherBL {
+
+    private static final Logger LOGGER = Logger.getLogger(GiftVoucherBL.class.getName());
+    private final TestExecutionContext context;
+    private final SoftAssertions softly;
+    private final String currentUserPersona;
+    private final Platform currentPlatform;
+    private static int totalAmount = 0;
+    private Map testData = Runner.getTestDataAsMap("indigoGiftVoucherDetails");
+    private static final Map details = new LinkedHashMap();
+
+    public GiftVoucherBL() {
+        long threadId = Thread.currentThread().getId();
+        this.context = Runner.getTestExecutionContext(threadId);
+        softly = Runner.getSoftAssertion(threadId);
+        this.currentUserPersona = SAMPLE_TEST_CONTEXT.ME;
+        this.currentPlatform = Runner.platform;
+
+        details.put(GIFT_VOUCHER_TEST_CONTEXT.FIRST_NAME, Randomizer.randomizeString(5));
+        details.put(GIFT_VOUCHER_TEST_CONTEXT.MESSAGE, Randomizer.randomizeString(15));
+        details.put(GIFT_VOUCHER_TEST_CONTEXT.LAST_NAME, Randomizer.randomizeString(5));
+        details.put(GIFT_VOUCHER_TEST_CONTEXT.PHONE, RandomStringUtils.randomNumeric(10));
+        details.put(GIFT_VOUCHER_TEST_CONTEXT.EMAIL, Randomizer.randomizeString(5) + "@" + Randomizer.randomizeString(5) + ".com");
+
+        Runner.setCurrentDriverForUser(this.currentUserPersona, this.currentPlatform, context);
+    }
+
+    public GiftVoucherBL previewAndPersonaliseGiftVoucher(String quantity, String denominations) {
+
+        LOGGER.info("Navigate to Gift Vouchers");
+        HomeScreen.get().navigateToGiftVouchers();
+
+        LOGGER.info("Preview and Personalise the "+quantity+" Gift Vouchers of "+denominations+" rupees ");
+        GiftVoucherScreen.get().selectQuantity(quantity).selectDenomination(denominations)
+                .personaliseGiftVoucher(details)
+                .previewGiftVoucher();
+
+        totalAmount = Integer.parseInt(quantity) * Integer.parseInt(denominations);
+
+        LOGGER.info("Verify total amount is "+totalAmount+" rupees");
+        assertThat(GiftVoucherScreen.get().getTotalAmount()).isEqualTo(this.totalAmount);
+
+        return this;
+    }
+
+    public GiftVoucherBL applyAPromoCode(String promoCode) {
+
+        String promo = "";
+        if(promoCode.equals("invalid")){
+            promo = Randomizer.randomizeAlphaNumericString(8);
+        }
+        else{
+            promo = promoCode;
+        }
+        LOGGER.info("Apply a "+promoCode+" promo code");
+        GiftVoucherScreen.get().applyPromoCode(promo);
+
+        softly.assertThat(GiftVoucherScreen.get().getPromoCodeMessage())
+                .as("Promo code message should be equal to expected message")
+                .isEqualTo((String) testData.get("invalidMessage"));
+
+        assertThat(GiftVoucherScreen.get().getTotalAmount())
+                .as("Total amount should be equal to expected amount after applying promo code")
+                .isEqualTo(totalAmount);
```

Comment:
One single get should be use in a function

Reply:
Done